### DISCUSSION TITLE
Anti-Aliasing

### DISCUSSION THREAD

**Discussion by oli414** [OpenRCT2/OpenGraphics#29] [General]
Anti-aliasing is a topic that needs additional research. In the original graphics of RCT2, some objects appear to have an outline, this is the result of anti-aliasing. According to Simon Foster, he eventually found a way to use anti-aliasing without it mixing with the background.

Check out the wiki page for more details:
https://github.com/OpenRCT2/OpenGraphics/wiki/Anti-Aliasing

And the definition of anti-aliasing:
https://github.com/OpenRCT2/OpenGraphics/wiki/Terminology#anti-aliasing

**My Theories**
I suspect that the outline was an unwanted side effect. Simon Foster mentioned that he eventually found a way to separate the object from the background to avoid anti-aliasing to create an outline. This would explain why it is most common on RCT1 assets.
That said, 

A unique instance of anti-aliasing are the track pieces. Tracks generally seem to have anti-aliasing but do not have the issue of showing a line between track elements. Given the lack of tools to help with the creation of the graphics at the time I suspect that those pixels were modified by hand.

**My Opinions**
Whilst the anti-aliasing may have been nothing more than a limitation at the time that was overcome for most RCT2 objects. I feel like the number of RCT1 graphics still present in RCT2 it does still make an important contribution to the overall style and nostalgia. Perhaps we should leave its inclusion up to the artist, replicate what each object had separately or set up guidelines for 

**Worth investigating**
1. Do all objects in RCT1 have an outline?
2. If there are objects that don't, does it appear to have anti-aliasing within the sprite, or was anti-aliasing disabled altogether?
3. Are there objects introduced by RCT2 that have the outline?
4. If point 2 is the case, are there RCT2 objects that appear to have anti-aliasing completely disabled?
5. How was anti-aliasing done in combination with a remappable color?

**Comment by oli414** [comment]
I have investigated a lot of the points I mentioned. 
https://github.com/OpenRCT2/OpenGraphics/wiki/Anti-Aliasing

![image](https://user-images.githubusercontent.com/2348094/144307367-df1b5d39-de6f-468a-8adf-44db226437ef.png)

There is a direct connection between RCT1 and RCT2 in terms of the outline we see. Check the research page for all the details.

My biggest question now is what do we do for OpenGraphics. 

I have 3 suggestions:
1. No outlines anywhere, the outline was an unwanted side effect in RCT1 that was resolved for RCT2.
2. Identify categories of objects where we always use outlines. Almost all track types are from RCT1 and have the outline, as such all track types should always have the outlines.
3. Just directly copy what the original object had, so the wonderland toadstool (RCT1) gets an outline but a pirate flag pole (RCT2) doesn't.

Going for option 1 would be the easiest and most consistent. But keep in mind that the outline does add to the "style" that RCT2 has, overall I'd say almost a third of all the objects in the game have the outline. So even though it is inconsistent, it does contribute a lot to the look of the objects.

My personal preference would go to option 2. Defining our own "rules" for when an outline is applicable or not, inspired by where we see outlines the most in the original graphics.

**Comment by Gymnasiast** [comment]
I think it would make most sense to stick with either an outline or no outline for everything, for visual consistency. After all, the objects are going to look different anyway, and I think the outline is not going to be the most obvious difference.

**Comment by oli414** [comment]
I'll try to get more data, as well as get a comparison between a scene with and without outlines so we can have some visuals of what it looks like.
